### PR TITLE
Put the webextension submission behind a waffle switch (bug #921)

### LINKED ISSUE #921: Put WebExtension submission behind a flag
I think the ability to submit add-ons as a WebExtension should be behind a flag, so that we can have it off in prod until we think through all the requirements.

Supports https://bugzilla.mozilla.org/show_bug.cgi?id=1210037

### PR DESCRIPTION
**Pull request by magopian**
Fixes #921

screen shot 2015-11-16 at 15 03 13

### CODE REVIEW

File: apps/files/tests/test_utils_.py [hunk 1]
```diff
@@ -133,8 +209,8 @@ def test_instanciate_without_data(self):
         with NamedTemporaryFile() as file_:
             file_.write(json.dumps(data))
             file_.flush()
-            mje = ManifestJSONExtractor(file_.name)
-            assert mje.data == data
+            pje = PackageJSONExtractor(file_.name)
+            assert pje.data == data
```
Comment (magopian) [inline apps/files/tests/test_utils_.py]: Already fixed that (I think) in a previous commit, but it was back, maybe a bad merge/rebase?

File: apps/devhub/utils.py [hunk 2]
```diff
@@ -146,7 +146,7 @@ def __init__(self, file_, addon=None, listed=None):
             # from the file itself.
             try:
                 addon_data = parse_addon(file_, check=False)
-            except ValidationError, form_error:
+            except ValidationError as form_error:
```
Comment (EnTeQuAk) [inline apps/devhub/utils.py]: grepping through the code we either use `as e` or `as error`, I personally prefer `as exc` but it seems `as e` wins regarding "most often used phrase"

File: apps/files/tests/test_utils_.py (hunk 1 above)
Comment (EnTeQuAk) [inline apps/files/tests/test_utils_.py]: I actually didn't find any other commit that changed that so maybe in some local branch?

File: apps/files/utils.py [hunk 3]
```diff
@@ -97,7 +97,10 @@ def parse(cls, path):
         elif os.path.exists(package_json):
             return PackageJSONExtractor(package_json).parse()
         elif os.path.exists(manifest_json):
-            return ManifestJSONExtractor(manifest_json).parse()
+            if waffle.switch_is_active('webextensions'):
+                return ManifestJSONExtractor(manifest_json).parse()
+            else:
+                raise forms.ValidationError("Webextensions aren't allowed yet")
```
Comment (EnTeQuAk) [inline apps/files/utils.py]: `WebExtensions` vs `Webextensions`?

Comment (andymckay-limited-access) [inline apps/files/utils.py]: WebExtensions pls

On Mon, Nov 16, 2015 at 2:41 PM, Christopher Grebs <notifications@github.com

File: apps/files/tests/test_utils_.py (hunk 1 above)
Comment (magopian) [inline apps/files/tests/test_utils_.py]: I think the fix was in a commit in a PR that was rebased... and lost at that moment. That's the only explanation I can think of.

**Comment by dbaron** [comment]
Hey, this created bogus comments in https://bugzilla.mozilla.org/show_bug.cgi?id=921 .  Could you try to avoid doing that?  (I'm not sure how the github robot decides when to comment in a bug, but it wasn't the right thing here.)

**Comment by magopian** [comment]
Thanks @dbaron for raising this issue, indeed it shouldn't have done that. I see two webhooks that may be related to this, maybe @andymckay and/or @kmaglione have an idea about that?

**Pull request merged**